R/callbacks-mixup.R

```diff
@@ -0,0 +1,111 @@
+#' Mixup callback
+#'
+#' Implementation of ['mixup: Beyond Empirical Risk Minimization'](https://arxiv.org/abs/1710.09412).
+#' As of today, tested only for categorical data,
+#' where targets are expected to be integers, not one-hot encoded vectors.
+#' This callback is supposed to be used together with [nn_mixup_loss()].
+#'
+#' @details
+#' Overall, we follow the fastai implementation
+#' (https://github.com/fastai/fastai/blob/master/fastai/callback/mixup.py)
+#' described here: https://forums.fast.ai/t/mixup-data-augmentation/22764).
+#' Namely,
+#' - We work with a single dataloader only, randomly mixing two observations from the same batch.
+#' - We linearly combine losses computed for both targets:
+#'   `loss(output, new_target) = weight * loss(output, target1) + (1-weight) * loss(output, target2)`
+#' - We draw different mixing coefficients for every pair.
+#' - We replace `weight` with `weight = max(weight, 1-weight)` to avoid duplicates.
+#'
+#' @param alpha parameter for the beta distribution used to sample mixing coefficients
+#'
+#' @examples
+#' mixup_callback <- luz_callback_mixup()
```

Comment:
Examples don't run on CRAN so we need to wrap them into `if (torch::torch_is_installed)` like eg: https://github.com/mlverse/luz/blob/main/R/lr-finder.R#L53-L65

Reply:
done!